### PR TITLE
feat: Add support for analytics disable module in `telemetry` dir

### PR DESCRIPTION
**Pull request by pgrzesik**
Needed for https://github.com/serverless/serverless/issues/9257 and refactoring to `telemetry` from `analytics`

### CODE REVIEW

File: disable-serverless-stats-requests.js [hunk 1]
```diff
@@ -6,13 +6,15 @@ const path = require('path');
 const isModuleNotFoundError = require('ncjsm/is-module-not-found-error');
 
 const resolveAreAnalyticsDisabled = (serverlessPath) => {
-  let modulePath = path.join(serverlessPath, 'lib/utils/analytics/areDisabled');
-  try {
-    return require.resolve(modulePath);
-  } catch (error) {
-    if (!isModuleNotFoundError(error, modulePath)) throw error;
+  for (const p of ['lib/utils/telemetry/areDisabled', 'lib/utils/analytics/areDisabled']) {
```
Comment (medikoo) [inline disable-serverless-stats-requests.js:9]: I'd keep the var name as `modulePath`. At least at first glance it's not clear what `p` could be, it's nice to keep it self explanatory

File: disable-serverless-stats-requests.js [hunk 2]
```diff
@@ -6,13 +6,15 @@ const path = require('path');
...
+    try {
+      return require.resolve(p);
+    } catch (error) {
+      if (!isModuleNotFoundError(error, p)) throw error;
```
Comment (medikoo) [inline disable-serverless-stats-requests.js:13]: I think we can also simplify it and simply assume that any error is module not found error (with `require.resolve` I'm not sure if it's possible to get other kind of error)

File: disable-serverless-stats-requests.js (hunk 1 above)
Comment (pgrzesik) [reply disable-serverless-stats-requests.js]: good call :+1:

File: disable-serverless-stats-requests.js (hunk 2 above)
Comment (pgrzesik) [reply disable-serverless-stats-requests.js]: I think it's a fine assumption - changed

File: disable-serverless-stats-requests.js [hunk 3]
```diff
@@ -3,16 +3,17 @@
...
 const resolveAreAnalyticsDisabled = (serverlessPath) => {
-  let modulePath = path.join(serverlessPath, 'lib/utils/analytics/areDisabled');
-  try {
-    return require.resolve(modulePath);
-  } catch (error) {
-    if (!isModuleNotFoundError(error, modulePath)) throw error;
+  for (const modulePath of ['lib/utils/telemetry/areDisabled', 'lib/utils/analytics/areDisabled']) {
+    try {
+      return require.resolve(modulePath);
+    } catch (error) {
```
Comment (medikoo) [inline disable-serverless-stats-requests.js:11]: nit: we may also strip `(error)` once not used

**Review by medikoo — APPROVED** [review]
👍

**Review by medikoo — APPROVED** [review]
👍

**Pull request merged**